cpmpy/solvers/minizinc.py

```diff
@@ -284,6 +287,22 @@ def solver_var(self, cpm_var) -> str:
             varname = cpm_var.name
             mzn_var = varname.replace(',', '_').replace('.', '_').replace(' ', '_').replace('[', '_').replace(']', '')
 
+            #test if the name is a valid minizinc identifier
+            #name must be of this form
+            regex = '^[A-Za-z][A-Za-z0-9_]*$'
```

Comment:
`solver_var()` is a hot path, I mean that it will be called a lot.

So, it would be better to compile the regex in the constructor and then use that here everytime instead.
https://docs.python.org/3/howto/regex.html#compiling-regular-expressions

Reply:
I added the compiled regex in the constructor
Keeping the assignment of keywords in solver_var() is ok?